SimulationAgent/DeviceState/DeviceStateActor.cs

```diff
@@ -183,7 +185,7 @@ private ISmartDictionary GetInitialState(DeviceModel model)
         {
             var initialState = CloneObject(model.Simulation.InitialState);
 
-            var state = new SmartDictionary(initialState);
+            var state = new SmartDictionary(initialState ?? new Dictionary<string, object>());
```

Comment:
please move the null check into SmartDictionary ctor, otherwise we'll have to nullcheck each time we use that class

Reply:
Done.